### PR TITLE
#38 Feature:Update profile 

### PR DESCRIPTION
**Pull request by gyssa-prince**
What does this PR do?

As **operator or driver**, I want to be able to update information on my profile, so that others could see updated information about me.

Description of Task to be completed?
-The operator and driver should be able to update information about their own profile.
-The operator and driver profile should be able accessible publicly.
-Appropriate error messages are displayed on unsuccessful attempts.

How should this be manually tested?

-Clone the repo
-checkout to branch ft-update-profile-#38
-Run npm install to install modules
-Run npm run dev:start to start the development server

Go to your postman
**To see the profile of user**
-Open postman and run POST http://localhost:3000/api/v1/profile/id of user
**To update the profile of user**
-Open postman and run POST http://localhost:3000/api/v1/profile/update/id of user
-Go to Body in your postman select JSON

put there anything you want to edit in only these info (name,email,phone,password)("name":"your name","phone":"your new phoneNo")
your will see a message which says "Profile updated" and status code of 200 if succeed

**To see if your information has been update**
-Open you postman and run POST http://localhost:3000/api/v1/profile/id of user  
or go to your database table and see if data has changed

What are the relevant Pivotal Tracker/trello stories?
https://trello.com/c/w1ftSCaI

**Screenshots**
Screen Shot 2022-06-17 at 09 31 24

### CODE REVIEW

**Review by Benafrica — APPROVED** [review]

File: src/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import dotenv from 'dotenv';
+import model from '../models';
```
Comment (songa1) [inline src/controllers/user.js:2]: This is raising errors and when there is an error, it keeps loading. So, please find a way to catch those errors.

File: src/routes/index.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ export default (app) => {
   app.post('/register', AuthController.signUp);
```
Comment (songa1) [inline src/routes/index.js:8]: Please make use of this file.

File: src/app.js [hunk 3]
```diff
@@ -31,15 +31,14 @@ server.get("/", (req, res) => {
...
+server.use(cookieParser())
 
 server.use("/api-docs", swaggerUi.serve, swaggerUi.setup(swaggerDoc, { explorer: true }));
-server.use("*", (req, res, next) => {
-	res.status(404).json({ error: "NOT FOUND", });
-});
+// server.use("*", (req, res, next) => {
+// 	res.status(404).json({ error: "NOT FOUND", });
+// });
+server.use("/profile",userRoute);
```
Comment (songa1) [inline src/app.js]: Do not put this here directly! Please use the file `/routers/index.js` to hold all of routes.

**Review by songa1 — CHANGES_REQUESTED** [review]
Check out some requested changes. And from what I have seen, you haven't written tests. If you did, there are some errors.

**Review by songa1 — COMMENTED** [review]
Please rebase and get recent changes.

**Review by songa1 — APPROVED** [review]

**Review by Stevenkwizera06 — APPROVED** [review]

**Review by Juleshb — APPROVED** [review]

**Review by arthurjunior250 — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -6,13 +6,15 @@
   "scripts": {
     "dev:start": " nodemon --exec babel-node src/index.js",
     "dev": "better-npm-run dev",
-    "test": "mocha --recursive --exit --timeout 1000000 --require @babel/register src/tests/*.test.js -watch",
+    "test": "mocha --recursive --exit --timeout 1000000 --require @babel/register src/tests/index.test.js ",
```
Comment (songa1) [inline package.json]: And please change this script. You are referring test located in index.test.js which we don't have.

You should be reffering to every file in /src/tests/

**Review by songa1 — CHANGES_REQUESTED** [review]
Please fix your tests. Even though you disabled them, they are even failing locally Refer to the resources I shared in the channel, and it would be better if you don't use real data while testing. Meaning that you don't need to use the development database while testing.

**Review by songa1 — APPROVED** [review]

**Pull request merged**